sycl/source/detail/plugin.hpp

```diff
@@ -259,6 +268,8 @@ class plugin {
   // represents the unique ids of the last device of each platform
   // index of this vector corresponds to the index in PiPlatforms vector.
   std::vector<int> LastDeviceIds;
+
+  bool MShutdownModeEnabled = false;
```

Comment:
Should this be thread-safe?
I imagine two threads: one is reading the flag and the other one is writing into it.